nncf/experimental/torch/nas/bootstrapNAS/elasticity/elastic_kernel.py

```diff
@@ -0,0 +1,495 @@
+            raise AttributeError('Invalid kernel size={} in scope={}.\nIt should be within the range: [1, {}]'.format(
+                kernel_size, self.node_name, self.max_kernel_size))
+
+        self._active_kernel_size = kernel_size
+
+    @property
+    def kernel_size_list(self) -> List[kernel_size_type]:
+        """
+        Gets list of all available kernel sizes to select from. Each value corresponds to a single element in the
+        search space of operation. The search space of the model is cartesian product of search spaces of operation.
+
+        :return: list of kernel sizes
+        """
+        return self._kernel_size_list
+
+    @property
+    def node_name(self) -> NNCFNodeName:
+        """
+        :return: node name that corresponds to operation with elastic kernel
+        """
+        return self._node_name
+
+    @property
+    def max_kernel_size(self) -> kernel_size_type:
+        """
+        :return: kernel size in the original operation
+        """
+        return self._max_kernel_size
+
+
+class ElasticKernelConv2DOp(ElasticKernelOp, nn.Module):
+    """
+    Introduces elastic kernel for the 2D convolution. On the forward pass it takes parameters of operations
+    and modifies in the way that kernel size is changing to a given value.
+    """
+
+    def __init__(self, max_kernel_size: kernel_size_type, node_name: NNCFNodeName,
+                 elastic_kernel_params: Optional[Dict[str, Any]]):
+        """
+        Constructor.
+
+        :param max_kernel_size: maximum kernel size value in the original operation.
+        :param node_name: string representation of operation address. It's used for more informative messages only.
+        :param elastic_kernel_params: parameters to configure elastic kernel for the operation.
+        """
+        super().__init__(max_kernel_size=max_kernel_size, node_name=node_name)
+        self._max_num_params = elastic_kernel_params.get('max_num_kernels', -1)
+        # Create kernel_size_list based on max module kernel size
+        self._kernel_size_list = self.generate_kernel_size_list(max_kernel_size)
+        self._ks_set = list(set(self.kernel_size_list))
+        self._ks_set.sort()
+
+        scale_params = {}
+        for i in range(len(self._ks_set) - 1):
+            ks_small = self._ks_set[i]
+            ks_larger = self._ks_set[i + 1]
+            param_name = '%dto%d' % (ks_larger, ks_small)
+            # noinspection PyArgumentList
+            scale_params['%s_matrix' % param_name] = Parameter(torch.eye(ks_small ** 2))
+        for name, param in scale_params.items():
+            self.register_parameter(name, param)
+
+    def generate_kernel_size_list(self, max_kernel_size: kernel_size_type) -> List[kernel_size_type]:
+        """
+        Generates list of available kernel size values.
+
+        :param max_kernel_size: maximum value of kernel size
+        :return: list of kernel size values.
+        """
```

Comment:
Please specify that kernel size should be odd and that default kernel value step is 2

Reply:
Thanks! I've specified